invenio_swh/service.py

```diff
@@ -60,35 +58,37 @@ def result_item(self, deposit: SWHDeposit):
         """Return a result item."""
         return self.result_cls(deposit)
 
-    def __init__(self, swh_controller: SWHController):
+    def __init__(self, controller: SWHController):
         """Instantiate the service.
 
         Injects the software heritage controller into the service.
         """
-        self.swh_controller = swh_controller
+        self.controller = controller
 
     @unit_of_work()
     def create(self, record, uow=None):
         """Create a new deposit.
 
-        If the controller fails to create the deposit, the transaction will be rolledback by the Unit of Work
-        and the deposit won't be created locally.
+        If the controller fails to create the deposit, it won't be created locally either.
         """
         self.validate_record(record)
 
-        deposit = self.record_cls.create(record.id)
-
         metadata = self.schema.dump(record)
-        swh_deposit = self.swh_controller.create_deposit(metadata)
-        deposit.model.swh_deposit_id = str(swh_deposit["deposit_id"])
-        deposit.model.status = SWHDepositStatus.CREATED
+        swh_deposit = self.controller.create_deposit(metadata)
+        deposit_id = swh_deposit.get("deposit_id")
+        if not deposit_id:
+            raise DepositNotCreated("Deposit id not returned by SWH.")
+
+        deposit = self.record_cls.create(record.id)
```

Comment:
Moved `deposit.create` after the deposition is created on remote, otherwise we could end up with a deposition locally that doesn't exist in SWH.